docs: specify Include Original Self Describing Event (#548)
snowplow/documentation: docs/destinations/forwarding-events/google-tag-manager-server-side/snowplow-client-for-gtm-ss/snowplow-client-configuration/index.md

By default, the self describing event will be "shredded" into a key using the schema name as the key, this is a "lossy" transformation, as the Minor and Patch parts of the jsonschema version will be dropped. This flag populates the original, lossless, Self Describing Event as `x-sp-self_describing_event`.

Let's assume we have a self-describing event following the schema `iglu:com.acme/foobar/jsonschema/1-0-0`. By default, the option to *Include Original Self Describing Event* is disabled. So the Snowplow client by default will include it in the common event as:

```
'x-sp-self_describing_event_com_acme_foobar_1': {
    foo: 'bar'
}
```

In case the option to *Include Original Self Describing Event* is enabled, then the Snowplow client, if it finds the original event (see note below), will also include it in the common event, resulting in:

```
'x-sp-self_describing_event_com_acme_foobar_1': {
    foo: 'bar'
},
'x-sp-self_describing_event': {
    schema: 'iglu:com.acme/foobar/jsonschema/1-0-0',
    data: {
        foo: 'bar'
    }
}
```

> Please note that this option only makes sense in using GTM as [**Server Side Tag Manager (pre-pipeline)**](https://docs.snowplow.io/docs/destinations/forwarding-events/google-tag-manager-server-side/#configuration-options) architecture because it only makes a difference when the input is a raw Snowplow event. In a [**Destinations Hub (post-pipeline)**](https://docs.snowplow.io/docs/destinations/forwarding-events/google-tag-manager-server-side/#configuration-options) architecture, this **option does not apply**. There — no matter how this option is configured — it is possible to access/filter data based only on self-describing data. In the above example, Destinations Hub would only contain 'x-sp-self_describing_event_com_acme_foobar_1', while the Server Side Tag Manager would include `x-sp-self_describing_event` in case the *Include Original Self Describing Event* is activated. 

Reviewer: ```suggestion
:::note

This option only makes sense when using GTM in a [**Server Side Tag Manager (pre-pipeline)**](/docs/destinations/forwarding-events/google-tag-manager-server-side/index.md#configuration-options) architecture, because it only makes a difference when the input is a _raw_ Snowplow event.

In a [**Destinations Hub (post-pipeline)**](/docs/destinations/forwarding-events/google-tag-manager-server-side/index.md#configuration-options) architecture, this option **does not apply**. Effectively, it’s always disabled, regardless of the setting. In the example above, this would mean that the data will contain `x-sp-self_describing_event_com_acme_foobar_1`, but not`x-sp-self_describing_event`.

:::
```

Reviewer: See https://github.com/snowplow/documentation#links and https://github.com/snowplow/documentation#formatting-content, also edited for clarity

By default, the self describing event will be "shredded" into a key using the schema name as the key, this is a "lossy" transformation, as the Minor and Patch parts of the jsonschema version will be dropped. This flag populates the original, lossless, Self Describing Event as `x-sp-self_describing_event`.

Let's assume we have a self-describing event following the schema `iglu:com.acme/foobar/jsonschema/1-0-0`. By default, the option to *Include Original Self Describing Event* is disabled. So the Snowplow client by default will include it in the common event as:

> ```
> 'x-sp-self_describing_event_com_acme_foobar_1': {
>     foo: 'bar'
> }
> ```

Reviewer: Actually, is this supposed to be JSON? If so:
````suggestion
```json
"x-sp-self_describing_event_com_acme_foobar_1": {
    "foo": "bar"
}
```
````

In case the option to *Include Original Self Describing Event* is enabled, then the Snowplow client, if it finds the original event (see note below), will also include it in the common event, resulting in:

```
'x-sp-self_describing_event_com_acme_foobar_1': {
    foo: 'bar'
},
'x-sp-self_describing_event': {
    schema: 'iglu:com.acme/foobar/jsonschema/1-0-0',
    data: {
        foo: 'bar'
    }
}

> ```

Reviewer: Same

> ```
> 'x-sp-self_describing_event_com_acme_foobar_1': {
>     foo: 'bar'
> },
> 'x-sp-self_describing_event': {
>     schema: 'iglu:com.acme/foobar/jsonschema/1-0-0',
>     data: {
>         foo: 'bar'
>     }
> }
> ```

Reviewer: ````suggestion
```json
"x-sp-self_describing_event_com_acme_foobar_1": {
    "foo": "bar"
},
"x-sp-self_describing_event": {
    "schema": "iglu:com.acme/foobar/jsonschema/1-0-0",
    "data": {
        "foo": "bar"
    }
}
```
````
